File: server/src/main/java/com/Ascendia/server/entity/Project/Project.java
package com.Ascendia.server.entity.Project;

import com.Ascendia.server.entity.Administrator.User;
import com.Ascendia.server.entity.ProjectManager.Task;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.time.LocalDate;

@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
@Entity
@Table(name = "project")
public class Project {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long projectId; // Change the type to Long or Integer, as per your preference

    @Column(name = "project_name", nullable = false)
    private String projectName; // Updated field name to follow Java naming conventions

    @Column(name = "project_type", nullable = false)
    private String projectType;

    @Column(name = "project_description", nullable = false, length = 1000)
    private String projectDescription;

    @Column(name = "project_status", nullable = false)
    private String projectStatus;

    @Column(name = "created_date", nullable = false)
    private LocalDate createdDate;

    @Column(name = "end_date", nullable = false)
    private LocalDate endDate;

    @Column(name = "image")
    private String image;

    @ManyToOne
    @JoinColumn(name = "pmanager_id", referencedColumnName = "userID")
    private User projectManager;

    @Column(name = "progress", nullable = false)
    private double progress; // New field for project progress

    @Column(name = "active", nullable = false)
    private boolean active = true;


    public Project(String projectName, String projectType, String location, String projectDescription, String projectStatus, LocalDate createdDate, LocalDate endDate, String image, User projectManager, double progress) {
        this.projectName = projectName;
        this.projectType = projectType;
        this.projectDescription = projectDescription;
        this.projectStatus = projectStatus;
        this.createdDate = createdDate;
        this.endDate = endDate;
        //this.pmId = pmId;
        this.image = image;
        this.projectManager = ProjectManager;
        this.progress = progress; // Initialize the progress field
    }
}

File: Mobile Client/lib/SiteEngineer/Project.dart
import 'dart:async';
import 'dart:convert';
//import 'dart:ffi';

import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;
import 'package:my_project/SiteEngineer/User.dart';

class Project {
  final int projectId;
  final String projectName;
  final String projectType;
  final String projectDescription;
  final String projectStatus;
  final DateTime createdDate;
  final DateTime endDate;
  final User projectManager;
  final String image;

  Project({
    required this.projectId,
    required this.projectName,
    required this.projectType,
    required this.projectDescription,
    required this.projectStatus,
    required this.createdDate,
    required this.endDate,
    required this.projectManager,
    required this.image,
  });

  factory Project.fromJson(Map<String, dynamic> json) {
    return Project(
      projectId: (json['projectId']??0)as int,
      projectName: json['projectName']??'',
      projectType: json['projectType']??'',
      projectDescription: json['projectDescription']??'',
      projectStatus: json['projectStatus']??'',
      createdDate: DateTime.parse(json['createdDate']??DateTime.now().toIso8601String()),
      endDate: DateTime.parse(json['endDate']??DateTime.now().toIso8601String()),
      projectManager: User.fromJson(json['pmId']??{}),
      image: json['image']??'',
    );
  }
  Map<String, dynamic> toJson() => {
    'projectId': projectId,
    'projectName': projectName,
    'projectType': projectType,
    'projectDescription': projectDescription,
    'projectStatus': projectStatus,
    'createdDate': createdDate.toIso8601String(),
    'endDate': endDate.toIso8601String(),
    'projectManager': projectManager.toJson(),
    'image': image,
  };

}